src/renderer/runner.ts

```diff
@@ -182,7 +194,10 @@ export class Runner {
       return RunResult.INVALID;
     }
 
-    const isReady = getIsDownloaded(version, localPath);
+    const isReady =
+      state === InstallState.installed ||
+      state === InstallState.downloaded ||
+      isValidBuild;
```

Comment:
Nope, I can't think of a such case where they will be different.
This is because - everytime the version is changed, it triggers [this](https://github.com/aryanshridhar/fiddle/blob/Integrate-fiddlecore/src/renderer/state.ts#L373-L376) event which updates the state of `currentElectronVersion`.